provider.go

```diff
@@ -38,25 +40,88 @@ func Provider(configfn schema.ConfigureFunc) terraform.ResourceProvider {
 	}
 }
 
+// Get the default value from the first function that returns one without an error
+func firstOf(key string, dvs ...schema.SchemaDefaultFunc) schema.SchemaDefaultFunc {
+	return func() (interface{}, error) {
+		for _, dv := range dvs {
+			if cfg, err := dv(); cfg != nil && err == nil {
+				return cfg, nil
+			}
+		}
+
+		return nil, fmt.Errorf("Unable to retrieve default for '%s' from anywhere", key)
+	}
+}
+
+// Look for default values in the BMC configuration file
+func fileBasedConfig(file string, key string) schema.SchemaDefaultFunc {
+	return func() (interface{}, error) {
+		cfg, err := ini.InsensitiveLoad(file)
+		if cfg != nil && err == nil {
+			sec, err := cfg.GetSection("")
+			if sec != nil && err == nil {
+				if sec.HasKey(key) {
+					v, err := sec.GetKey(key)
+					if v != nil && err == nil {
+						return v.String(), nil
+					}
+					return nil, fmt.Errorf("Key (%s) has no value in %s", key, file)
+				}
+				return nil, fmt.Errorf("Key (%s) not found in %s", key, file)
+			}
+			return nil, fmt.Errorf("No default section available in %s", file)
+		}
+		return nil, fmt.Errorf("Unable to load INI file (%s)", file)
+	}
+}
+
 func schemaMap() map[string]*schema.Schema {
 	return map[string]*schema.Schema{
 		"tenancy_ocid": {
 			Type:        schema.TypeString,
 			Required:    true,
 			Description: descriptions["tenancy_ocid"],
-			DefaultFunc: schema.EnvDefaultFunc("OBMCS_TENANCY_OCID", nil),
+			DefaultFunc: firstOf(
+				"tenancy_ocid",
+				schema.EnvDefaultFunc("OBMCS_TENANCY_OCID", nil),
+				fileBasedConfig(os.Getenv("HOME") + "/.oraclebmc/config", "tenancy")),
```

Comment:
you would want to use `path.Join` for these and define `.oraclebmc/config` as a const.